### ISSUE TITLE
Validation of src indentation for network/eos/eos_template

### ISSUE THREAD

**Issue by grybak-arista** [ansible/ansible-modules-core#4590]
##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

network/eos/eos_template.py
##### ANSIBLE VERSION

```

```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

When passing a src template into network/eos/eos_template, it would be nice to have the indentation of the template validated, and a failure generated if the indentation does not match the expected 3-space indent.

Often, a src template may be written using an editor with an indent set to something other than 3 spaces. We have seen cases where such an improperly indented src template does not cause a failure during playbook execution, but the final switch configuration does not end up as the customer expected. In other cases, the playbook may successfully set a valid configuration on the first run, but fail on subsequent runs because properly indented lines are removed, but improperly indented lines are re-applied without the needed block header command. In this instance, the switch reports one of the lines is an invalid input, but the line looks correct to the user.

Validation of the indentation would provide better feedback to diagnose these issues.
##### STEPS TO REPRODUCE

Run a playbook with an eos_template task using the following 2-space indented configuration as the src parameter (MY_ACL does not exist on switch):

```
ip access-list MY_ACL
  10 permit tcp host 172.31.33.250 any
  86 permit icmp any any
  90 deny ip any any
  exit
```

##### EXPECTED RESULTS

The playbook should fail with an error indicating the src configuration contained indentation other than 3 spaces.
##### ACTUAL RESULTS

Playbook may succeed without any errors, or it may fail indicating that one of the improperly indented lines is an invalid command.

In the above example, the first pass succeeds and does set the ACL properly, but the second pass fails stating "msg": "matched error in response: 10 permit tcp host 172.31.33.250 any\r\n% Invalid input" because the "ip access-list MY_ACL" command was not sent in the second pass (as expected).

```

```

**Comment by ansibot** [comment]
@privateip, @gundalow, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gundalow** [comment]
Hi,
Thanks for the suggestion. This sounds like something useful to add that could catch issues. We are currently busy working toward to final 2.2 release. After that release we can look at this.

**Comment by ansibot** [comment]
@grybak-arista We appreciate you taking the time to file this issue, but the module you've opened the issue about has been deprecated.

We will consider critical bugfixes for deprecated modules, but all other types of requests will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
